### PR TITLE
Ppd 251 gps integration test

### PR DESCRIPTION
**Pull request by alessio-acitelli**
image

#### List of Changes

- fixed and added integration tests for gps

#### Motivation and Context

- increased API coverage for the application

#### How Has This Been Tested?

local run

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/application-local.properties [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # info
+info.application.name=
+info.application.version=
+info.application.description=
```
Comment (jacopocarlini) [inline src/main/resources/application-local.properties]:
```suggestion
```

File: src/main/resources/application-dev.properties [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# info
+info.application.name=@project.artifactId@
+info.application.version=@project.version@
+info.application.description=@project.description@
```
Comment (jacopocarlini) [inline src/main/resources/application-dev.properties]:
```suggestion
```

File: src/main/java/it/gov/pagopa/spontaneouspayment/controller/impl/BaseController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @Value("${info.application.version}")
+    private String version;
+
+    @Value("${info.properties.environment}")
+    private String environment;
+
+    /**
+     * @return 200 OK
+     */
+    @Hidden
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/spontaneouspayment/controller/impl/BaseController.java]: If is hidden from the swagger [this section](https://github.com/pagopa/pagopa-spontaneous-payments/pull/25/files#diff-0590b599497e4f112969e688eac36f64d755109efaa89fcd289f9b07b3961b1aR41-R48) is not needed

File: integration-test/features/organizations.feature [hunk 4]
```diff
@@ -42,3 +42,7 @@ Feature: All about Organizations
     When the organization set the status to "DISABLED"
     Then the organization gets the status code 200
     And the status is "DISABLED" in the organization's details
+    
+  Scenario: An organization try to create an enrollment to a service without some mandatory parameters
+    Given the organization "777777" with the service "service-1" without mandatory parameters
```
Comment (jacopocarlini) [inline integration-test/features/organizations.feature]: missing the when statements. Use the When to make the action to test

File: integration-test/features/payments.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  # Case OK
+  Scenario: An organization creates a spontaneous payment
+    Given the organization "777777" with an enrollment to service "service-1"
+    When the organization creates a spontaneous payment
+    Then the organization gets the created status code 201 
+    And the organization gets the created Payment Position
+    
+  # Case KO: 404 non-existent service
+	  Scenario: An organization tries to create a spontaneous payment on a service that doesn't exist
+	  Given the organization "777777" with an enrollment to service "service-not-found"
```
Comment (jacopocarlini) [inline integration-test/features/payments.feature]: ditto: 
  missing the when statements. Use the When to make the action to test

**Review by jacopocarlini — COMMENTED** [review]
Good job. 👍 
I have only some observations

File: integration-test/features/organizations.feature [hunk 6]
```diff
@@ -42,3 +42,8 @@ Feature: All about Organizations
...
+    When the organization "777777" creates a creditor institution with enrollment to service
```
Comment (pasqualespica) [inline integration-test/features/organizations.feature]: the same parameter should not be repeated between steps.
```
    Given the organization "777777" with the service "service-1" without mandatory parameters
    When the organization "777777" creates a creditor institution with enrollment to service
```
in to something like
```
    Given the organization "777777"
    When the organization enrolls to service "service-1" without mandatory parameters
```

or better using in this case `Scenario Outline` with `Example`

File: integration-test/features/payments.feature [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  # Case KO: 404 non-existent service
+	  Scenario: An organization tries to create a spontaneous payment on a service that doesn't exist
+	  Given the organization creates the creditor institution "777777"
+	  When the organization enrolls the creditor institution on the service "service-not-found" 
+	  Then the organization gets the created status code 404
+  
+  # Case KO: 400 bad request - non-existent property in service configuration
+	  Scenario: An organization tries to create a spontaneous payment with a not configured property request
+	  Given the organization "777777" with an enrollment to service "service-1"
+	  When the organization creates a spontaneous payment with a not configured property in the request
```
Comment (pasqualespica) [inline integration-test/features/payments.feature]: change this 
`When the organization creates a spontaneous payment with a not configured property in the request`
in 
`When the organization creates a spontaneous payment without <PROP> property in the request`

File: integration-test/features/payments.feature [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    Given GPS running
+
+  # Case OK
+  Scenario: An organization creates a spontaneous payment
+    Given the organization "777777" with an enrollment to service "service-1"
+    When the organization creates a spontaneous payment
+    Then the organization gets the created status code 201 
+    And the organization gets the created Payment Position
+    
+  # Case KO: 404 non-existent service
```
Comment (pasqualespica) [inline integration-test/features/payments.feature:12]: and the case that organization does't exist ( or is not enabled )  ?

**Review by pasqualespica — CHANGES_REQUESTED** [review]
Thera are several duplicated features ex. `integration-test/features/organizations.feature`
```
  Scenario: An organization change the IBAN
    Given the organization "777777" with the service "service-1"
    When the organization changes the service IBAN with "ABCD"
    Then the organization gets the status code 200
    And the service for the organization has the IBAN "ABCD"
  Scenario: An organization change the IBAN
    Given the organization "777777" with the service "service-1"
    When the organization changes the service IBAN with "ABCD"
    Then the organization gets the status code 200
    And the service for the organization has the IBAN "ABCD"
``` 

Can you check this cases ?

File: integration-test/features/payments.feature (hunk 8 above)
Comment (alessio-acitelli) [reply integration-test/features/payments.feature:12]: At the moment we do not manage the enabled / disabled cases

File: integration-test/features/payments.feature (hunk 7 above)
Comment (alessio-acitelli) [reply integration-test/features/payments.feature]: The test verifies that an error occurs when a property that is not configured on the service present in the db is passed.
Maybe I could change to:

When the organization tries to create a spontaneous payment with a "property-not-found" that is not expected in the service configuration

Comment (pasqualespica) [reply integration-test/features/payments.feature]: or something like that `When the organization tries to create a spontaneous payment with a property "property-not-found" not configured to service`

**Review by pasqualespica — APPROVED** [review]
great 👍

**Pull request merged**